src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -1538,20 +1538,52 @@ namespace shogun
 		}
 
 		/**
-		 * Method that computes the standard deviation of vectors or matrices
+		 * Method that computes the standard deviation of matrices
 		 * composed of real numbers.
 		 *
-		 * @param a SGVector or SGMatrix
-		 * @return The vector mean \f$\bar{a}_i\f$ or matrix mean
-		 * \f$\bar{m}_{i,j}\f$
+		 * @param mat the SGMatrix
+		 * @param featurewise whether to calculate featurewise or for whole
+		 * matrix
+		 * @return The standard deviation
 		 */
-		template <typename T>
-		typename std::enable_if<
-		    !std::is_same<T, complex128_t>::value, SGVector<float64_t>>::type
-		std_deviation(const SGMatrix<T>& mat, bool colwise = true)
+		template <
+		    typename T, typename std::enable_if_t<
+		                    !std::is_same<T, complex128_t>::value>* = nullptr>
+		SGVector<T>
+		std_deviation(const SGMatrix<T>& mat, bool featurewise = true)
 		{
 			REQUIRE(mat.size() > 0, "Vector/Matrix cannot be empty!\n");
-			return infer_backend(mat)->std_deviation(mat, colwise);
+			return infer_backend(mat)->std_deviation(mat, featurewise);
+		}
+
+		/**
+		 * Method that computes the median of matrices
+		 * composed of real numbers.
+		 *
+		 * @param a SGVector or SGMatrix
+		 * @return The whole matrix median
+		 */
+		template <
+		    typename T, template <typename> class Container,
+		    typename std::enable_if_t<!std::is_same<T, complex128_t>::value>* =
+		        nullptr>
+		T median(const Container<T>& a)
```

Comment:
some food for thought on fast median computation: https://www.stat.cmu.edu/~ryantibs/papers/median.pdf

Reply:
I guess we can add the option when the exact value isn't required? I am just thinking that the stl implementation must be quite efficient already no?

Reply:
Yes just wanted to share it